src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -1350,33 +1350,37 @@ boolean classes(boolean handleExceptions, boolean defineAdapters, boolean conver
             out.println("#ifdef SHARED_PTR_NAMESPACE");
             out.println("template<class T> class SharedPtrAdapter {");
             out.println("public:");
-            out.println("    typedef SHARED_PTR_NAMESPACE::shared_ptr<T> S;");
-            out.println("    SharedPtrAdapter(const T* ptr, size_t size, void* owner) : ptr((T*)ptr), size(size), owner(owner),");
-            out.println("            sharedPtr2(owner != NULL && owner != ptr ? *(S*)owner : S((T*)ptr)), sharedPtr(sharedPtr2) { }");
-            out.println("    SharedPtrAdapter(const S& sharedPtr) : ptr(0), size(0), owner(0), sharedPtr2(sharedPtr), sharedPtr(sharedPtr2) { }");
-            out.println("    SharedPtrAdapter(      S& sharedPtr) : ptr(0), size(0), owner(0), sharedPtr(sharedPtr) { }");
-            out.println("    SharedPtrAdapter(const S* sharedPtr) : ptr(0), size(0), owner(0), sharedPtr(*(S*)sharedPtr) { }");
-            out.println("    void assign(T* ptr, size_t size, void* owner) {");
+            out.println("    typedef typename SHARED_PTR_NAMESPACE::remove_const<T>::type TU;");
```

Comment:
Either these `remove_const`, `add_const`, or we must ensure in `adapterInformation` that `valueTypeName` has no const (for all adapters ??).